### PR TITLE
FEATURE: Reserved usernames

### PR DESCRIPTION
**Pull request by riking**
basically this https://twitter.com/FioraAeterna/status/614433067877273601

### CODE REVIEW

**Comment by eviltrout** [comment]
This really needs a test

**Comment by riking** [comment]
Yep, seeing as I typoed the site setting name.

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -231,6 +231,10 @@ def create
       return fail_with("login.password_too_long")
     end
 
+    if SiteSetting.reserved_usernames.include? params[:username].downcase
+      return fail_with("login.reserved_username")
+    end
+
```
Comment (nlalonde) [inline app/controllers/users_controller.rb]: This should be validated in the model. See username_validator in user.rb.

Comment (riking) [inline app/controllers/users_controller.rb]: Nope, that will add problems for the existing @discourse account created on the hosted sites, and you wouldn't ever be able to add a username to the default list without causing people problems.

**Comment by coding-horror** [comment]
Agreed we should pull this in

**Comment by eviltrout** [comment]
Bringing it in.

**Pull request merged**